pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroup.java

```diff
@@ -99,6 +104,7 @@ protected ResourceGroup(ResourceGroupService rgs, String rgName,
     public ResourceGroup(ResourceGroup other) {
         this.resourceGroupName = other.resourceGroupName;
         this.rgs = other.rgs;
+        this.resourceGroupPublishLimiter = other.resourceGroupPublishLimiter;
         this.setResourceGroupMonitoringClassFields();
 
         // ToDo: copy the monitoring class fields, and ruPublisher/ruConsumer from other, if required.
```

Comment:
Don't we also need an update of the resourceGroupPublishLimiter at an update of the RG (in updateResourceGroup())?

Reply:
Yes. But I was hoping that adjusting the limiters based on usages and whenever resource group is updated,  can be taken up in the next commit.

Reply:
OK. I will try and add an update to the publish rate-limiter when the RGs are updated (together with the periodic adjustment based on other brokers' reports).